newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/ErrorEventFactory.java

```diff
@@ -166,4 +187,22 @@ private static Map<String, Object> buildUserAttributes(String appName, Transacti
         }
         return userAttributes;
     }
+
+    private static String invokeErrorGroupCallback(ErrorData errorData) {
+        String groupId = null;
+
+        try {
+            if (ErrorGroupCallbackHolder.getErrorGroupCallback() != null) {
+                long start = System.currentTimeMillis();
+                groupId = ErrorGroupCallbackHolder.getErrorGroupCallback().generateGroupingString(errorData);
+                long duration = System.currentTimeMillis() - start;
+                NewRelic.getAgent().getMetricAggregator().recordResponseTimeMetric(MetricNames.SUPPORTABILITY_ERROR_GROUPING_CALLBACK_EXECUTION_TIME, duration);
+                Agent.LOG.finest(MessageFormat.format("Customer errorGroupCallback generated groupId of [{0}] in {1}ms", groupId, duration));
+            }
+        } catch (Exception e) {
+            Agent.LOG.warning(MessageFormat.format("Customer errorGroupCallback implementation threw an exception: {0}", e.getMessage()));
```

Comment:
Also note that the MessageFormat.format is executed even if the log is set to SEVERE only.